src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+            'choices' => [
+                'single_user' => static::LOGIN_MODE_SINGLE_USER,
+                'dynamic_user_creation' => static::LOGIN_MODE_DYNAMIC_USER,
+            ],
+            'attr' => [
+                'class' => 'dependent-trigger',
+                'data-dependent-group' => static::DEPENDENT_GROUP_LOGIN_MODE,
+            ],
+            'constraints' => new NotBlank(),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            $options
+        ));
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyUserField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(
+            PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER,
+            SelectType::class,
+            array_merge($this->getCompanyUserFieldOptions(), $options)
+        );
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitFieldListeners(FormBuilderInterface $builder)
+    {
+        $formModificationCallback = function (FormEvent $event) {
+            $form = $event->getForm();
+            $loginMode = $event->getData()[PunchoutCatalogConnectionSetupTransfer::LOGIN_MODE] ?? null;
+
+            if ($loginMode !== static::LOGIN_MODE_DYNAMIC_USER) {
+                $this->disableCompanyBusinessUnitField($form);
+
+                return;
+            }
+
+            $idParentCompanyBusinessUnit = $form
+                ->getParent()
+                ->getParent()
+                ->get(PunchoutCatalogConnectionTransfer::FK_COMPANY_BUSINESS_UNIT)
+                ->getData();
+
+            if (!$idParentCompanyBusinessUnit) {
+                return;
+            }
+
+            $this->updateCompanyBusinessUnitFieldChoices($form, $idParentCompanyBusinessUnit);
+        };
+
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, $formModificationCallback);
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, $formModificationCallback);
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addCompanyUserFieldListeners(FormBuilderInterface $builder)
+    {
+        $formModificationCallback = function (FormEvent $event) {
```

Comment:
Please see my comment to the similar case above. Create & Register - 2 actions in one method.